### PR TITLE
8333675: [lworld] Handling of invalid class files

### PR DESCRIPTION
**Pull request by fparain**
Improve detection of invalid class files and test coverage of invalid scenarios.

Tested with Mach5, tier 1 to 3.

---------
### Progress
- [x] Change must not contain extraneous whitespace

### Issue
 * [JDK-8333675](https://bugs.openjdk.org/browse/JDK-8333675): [lworld] Handling of invalid class files (**Bug** - P3)

### Reviewers
 * [David Simms](https://openjdk.org/census#dsimms) (@MrSimms - Committer)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/valhalla.git pull/1114/head:pull/1114` \
`$ git checkout pull/1114`

Update a local copy of the PR: \
`$ git checkout pull/1114` \
`$ git pull https://git.openjdk.org/valhalla.git pull/1114/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 1114`

View PR using the GUI difftool: \
`$ git pr show -t 1114`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/valhalla/pull/1114.diff">https://git.openjdk.org/valhalla/pull/1114.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/valhalla/pull/1114#issuecomment-2161337524)

### CODE REVIEW

File: test/hotspot/jtreg/runtime/valhalla/inlinetypes/classfileparser/ValueClassValidation.java [hunk 1]
```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (c) 2024, 2024, Oracle and/or its affiliates. All rights reserved.
```
Comment (dholmes-ora) [inline test/hotspot/jtreg/runtime/valhalla/inlinetypes/classfileparser/ValueClassValidation.java:2]: Remove one copyright year

Comment (fparain) [reply test/hotspot/jtreg/runtime/valhalla/inlinetypes/classfileparser/ValueClassValidation.java]: Fixed

**Review by MrSimms — APPROVED** [review]
Much improved test coverage.

**Comment by fparain** [comment]
Thank you for the reviews.

File: src/hotspot/share/classfile/classFileParser.cpp [hunk 2]
```diff
@@ -4728,13 +4728,19 @@ void ClassFileParser::verify_legal_class_modifiers(jint flags, const char* name,
...
       (is_interface && major_gte_1_5 && (is_identity || is_enum)) ||   //  ACC_SUPER (now ACC_IDENTITY) was illegal for interfaces
-      (!is_interface && major_gte_1_5 && is_annotation)) {
+      (!is_interface && major_gte_1_5 && is_annotation) ||
+      (!valid_value_class)) {
     ResourceMark rm(THREAD);
     const char* class_note = "";
+    if (!valid_value_class) {
+      class_note = " (a value class must be final or else abstract)";
+    }
     if (!is_identity)  class_note = " (a value class)";
```
Comment (dholmes-ora) [inline src/hotspot/share/classfile/classFileParser.cpp:4744]: @fparain you needed to delete line 4744 when incorporating my changes. Now you have overwritten the detailed message. I can fix that in my upcoming changes.

**Pull request closed without merging**